zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -747,22 +747,32 @@ private void setupClientPort() throws ConfigException {
             return;
         }
         QuorumServer qs = quorumVerifier.getAllMembers().get(serverId);
-        if (clientPortAddress != null && qs != null && qs.clientAddr != null) {
-            if ((!clientPortAddress.getAddress().isAnyLocalAddress() && !clientPortAddress.equals(qs.clientAddr)) || (
-                clientPortAddress.getAddress().isAnyLocalAddress()
-                && clientPortAddress.getPort() != qs.clientAddr.getPort())) {
-                throw new ConfigException("client address for this server (id = " + serverId
-                                          + ") in static config file is " + clientPortAddress
-                                          + " is different from client address found in dynamic file: " + qs.clientAddr);
-            }
+        if (qs == null) {
+            return;
         }
-        if (qs != null && qs.clientAddr != null) {
+        if (clientPortAddress == null && secureClientPortAddress == null && qs.clientAddr != null) {
             clientPortAddress = qs.clientAddr;
+            return;
         }
-        if (qs != null && qs.clientAddr == null) {
+        if (qs.clientAddr == null && clientPortAddress != null) {
             qs.clientAddr = clientPortAddress;
             qs.isClientAddrFromStatic = true;
+            return;
+        }
+        if (clientPortAddress != null
```

Comment:
I have to admit: I have no idea what or why you're checking this here.
Why are you bailing out here?
I think some clarifying comment would be helpful for these two if-statements.